Doc/library/stdtypes.rst

```diff
@@ -1583,6 +1583,12 @@ expression support in the :mod:`re` module).
          >>> 'Py' in 'Python'
          True
 
+      If the substring *sub* is ``''``, then `:meth:~str.find` method will return 
+      a value that is equal to ``len(str)``.
```

Comment:
I think the text should explain why.  As shows in the bpo-24243 ticket, people may be surprised by this behaviour if they don’t understand how it derives from Python’s notion of string slices.  If you know that `'' in 'abc'` is true + the doc of str.find says it returns the highest index of the found substring, then this is not a surprise.  I am not sure that mentioning `len` only here will help people make sense of it.

Reply:
Notation note: let’s write code like `len(s)`, not `len(str)`.